tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -105,13 +105,130 @@ void toPointCloud2(const TangoPointCloud& tango_point_cloud,
+    toLaserScanRange(x, y, z, min_height, max_height, laser_scan);
+  }
+  laser_scan->header.stamp.fromSec((tango_point_cloud.timestamp + time_offset) / 1e3);
+}
+// Converts a TangoPointCloud to a sensor_msgs::PointCloud2 and a
+// sensor_msgs::LaserScan.
+// @param tango_point_cloud, TangoPointCloud to convert.
+// @param time_offset, offset in ms between tango_point_cloud (tango time) and
+//        point_cloud/laser_scan (ros time).
+// @param min_height minimum height for a point of the point cloud to be
+// included in laser scan.
+// @param max_height maximum height for a point of the point cloud to be
+// included in laser scan.
+// @param point_cloud, the output PointCloud2.
+// @param laser_scan, the output LaserScan.
+void toPointCloud2AndLaserScan(const TangoPointCloud& tango_point_cloud,
+                   double time_offset,
+                   double min_height,
+                   double max_height,
+                   sensor_msgs::PointCloud2* point_cloud,
+                   sensor_msgs::LaserScan* laser_scan) {
+  point_cloud->width = tango_point_cloud.num_points;
+  point_cloud->height = 1;
+  point_cloud->point_step = (sizeof(float) * tango_ros_native::NUMBER_OF_FIELDS_IN_POINT_CLOUD);
+  point_cloud->is_dense = true;
+  point_cloud->row_step = point_cloud->width;
+  point_cloud->is_bigendian = false;
+  point_cloud->data.resize(tango_point_cloud.num_points);
+  sensor_msgs::PointCloud2Modifier modifier(*point_cloud);
+  modifier.setPointCloud2Fields(tango_ros_native::NUMBER_OF_FIELDS_IN_POINT_CLOUD,
+                                "x", 1, sensor_msgs::PointField::FLOAT32,
+                                "y", 1, sensor_msgs::PointField::FLOAT32,
+                                "z", 1, sensor_msgs::PointField::FLOAT32,
+                                "c", 1, sensor_msgs::PointField::FLOAT32);
+  modifier.resize(tango_point_cloud.num_points);
+  sensor_msgs::PointCloud2Iterator<float> iter_x(*point_cloud, "x");
+  sensor_msgs::PointCloud2Iterator<float> iter_y(*point_cloud, "y");
+  sensor_msgs::PointCloud2Iterator<float> iter_z(*point_cloud, "z");
+  sensor_msgs::PointCloud2Iterator<float> iter_c(*point_cloud, "c");
+  for (size_t i = 0; i < tango_point_cloud.num_points;
+      ++i, ++iter_x, ++iter_y, ++iter_z, ++iter_c) {
+    // Fill in point cloud message.
+    *iter_x = tango_point_cloud.points[i][0];
+    *iter_y = tango_point_cloud.points[i][1];
+    *iter_z = tango_point_cloud.points[i][2];
+    *iter_c = tango_point_cloud.points[i][3];
+
+    // Laser scan frame is rotated of 90 degrees around x axis with respect to
+    // point cloud frame.
+    double x = tango_point_cloud.points[i][0];
+    double y = tango_point_cloud.points[i][2];
+    double z = -tango_point_cloud.points[i][1];
+    toLaserScanRange(x, y, z, min_height, max_height, laser_scan);
+  }
+  point_cloud->header.stamp.fromSec((tango_point_cloud.timestamp + time_offset) / 1e3);
+  laser_scan->header = point_cloud->header;
```

Comment:
Yes, I certainly agree on that one (and it would be more maintainable too).
Even being the same order of magnitude, I don't know for sure if in this case it would affect performance to iterate twice. Pointclouds are pretty heavy as I understand. 

If iterating twice affected performance in a non-negligible way, it wouldn't be a bad idea to me to have two separate functions with their respective `for` loops as it is now. If the performance cost is negligible, I totally agree with you @mcopejans .